fem/gridfunc.hpp

```diff
@@ -877,6 +878,62 @@ class ExtrudeCoefficient : public Coefficient
 GridFunction *Extrude1DGridFunction(Mesh *mesh, Mesh *mesh2d,
                                     GridFunction *sol, const int ny);
 
+/// The class provides methods for projecting function values evaluated on a
+/// set of points to a grid function. The values are directly copied to the
+/// nodal values of the target grid function if any of the points is matching
+/// a node of the grid function. For example, if a parallel grid function is
+/// saved in parallel, every saved chunk can be read on every other process
+/// and mapped to a local grid function that does not have the same structure
+/// as the original one. The functionality is based on a kd-tree search in a
+/// cloud of points.
+class KDTreeNodalProjection
+{
+private:
+
+   /// Pointer to the KDTree for the 3D case
+   //KDTree3D* kdt3D;
+   std::unique_ptr<KDTree3D> kdt3D;
+
+   /// Pointer to the KDTree for the 2D case
+   //KDTree2D* kdt2D;
+   std::unique_ptr<KDTree2D> kdt2D;
+
+   /// Pointer to the target grid function
+   GridFunction* dest;
+
+   /// Upper corner of the bounding box
+   Vector maxbb;
+
+   /// Lower corner of the bounding box
+   Vector minbb;
+
+public:
+   /// The constructor takes as input an L2 or H1 grid function (it can be
+   /// a vector grid function). The Project method coppies a set of values
+   /// to the grid function.
+   KDTreeNodalProjection(GridFunction& dest_);
+
+   /// Frees the memory allocated for the transfer
+   ~KDTreeNodalProjection() {}
```

Comment:
Can remove this now and use the implicitly defined destructor

Reply:
resolved